source/PreferencesPanel.cpp

```diff
@@ -57,6 +57,7 @@ namespace {
 	const string SCROLL_SPEED = "Scroll speed";
 	const string FIGHTER_REPAIR = "Repair fighters in";
 	const string SHIP_OUTLINES = "Ship outlines in shops";
+	const string DISPLAY_ID = "Display ID (Nees restart.)";
```

Comment:
The last used display can be a "hidden" preference, rather than one that is important enough to be shown to users in the UI

Reply:
Not sure what do you mean by hidden preference... Can you point me to an example?

Reply:
```suggestion
	const string DISPLAY_ID = "Display ID (Needs restart)";
```

Reply:
> Not sure what do you mean by hidden preference... Can you point me to an example?

Not all preferences are visible from the in-game preferences menu. Some preferences must be changed by going to the preferences.txt file. For example, the window size that the game starts at is a preference not visible in game, meaning that if you wanted to manually change it (without opening the game and just dragging the edges of the window, that is) then you'd need to go to the preferences.txt file to do so.

And I agree with tehhowch's suggestion that the display ID should be a hidden preference. The game should remember what display the game was last on when it was closed, and therefore reopen in the same display, but you shouldn't need to be able to change the display that the game opens in from the in-game preferences menu; the method of choosing the display ID preference should be simply closing the game in the display you want the game to reopen in, just as choosing the window size is a matter of closing the game at the window size you want.

Reply:
Yep.

It'll require reacting to window move events and setting the display ID in the background, rather than by a field in the preferences panel.

Reply:
Ok, that is not simple, cause this does not get the display ID, it kinda starts on display 0 by default and you can change it in settings. Maybe SDL_GetDisplayBounds() can be used to find out which display I'm on before exit...

Reply:
Done!